python/lsst/faro/measurement/TractTableMeasurementTasks.py

```diff
@@ -144,3 +145,82 @@ def run(
                 metricName, np.mean(np.abs(result["corr"])), extras=extras
             )
         )
+
+
+class FluxStatisticConfig(MeasurementTaskConfig):
+    """Class to configure the flux statistic measurement task.
+    """
+
+    columns = DictField(
+        doc="""Columns required for metric calculation. These are full column names, encompassing all columns
+        in a SourceTable and columns that do not change name with band in an ObjectTable. If per-band columns
+        are required, use `columnsBand` instead.""",
+        keytype=str,
+        itemtype=str,
+        default={},
+    )
+    columnsBand = DictField(
+        doc="""Column suffixes used to identify the required columns for metric calculation. The band name
+        will be prepended to this stub to select the columns of interest. These values are used to select
+        flux columns in an objectTable.""",
+        keytype=str,
+        itemtype=str,
+        default={},
+    )
+    statistic = Field(
+        doc="Statistic name to use to generate the metric (from `~lsst.afw.math.Property`).",
+        dtype=str,
+        default="MEAN",
+    )
+    numSigmaClip = Field(
+        doc="Rejection threshold (sigma) for statistics clipping.",
+        dtype=float,
+        default=5.0,
+    )
+    clipMaxIter = Field(
+        doc="Maximum number of clipping iterations to apply.",
+        dtype=int,
+        default=3,
+    )
+
+
+class FluxStatisticTask(Task):
+    """Class to perform flux statistic calculations on parquet table data.
+
+    Output
+    ------
+    Flux metric with defined configuration.
+    """
+
+    ConfigClass = FluxStatisticConfig
+    _DefaultName = "FluxStatisticTask"
+
+    def run(
+        self, metricName, catalog, currentBands, **kwargs
+    ):
+
+        self.log.info("Measuring %s", metricName)
+
+        # filter catalog using selectors
+        catalog = selectors.applySelectors(catalog,
+                                           self.config.selectorActions,
+                                           currentBands=currentBands)
+
+        # extract flux value column
+        all_columns = [x for x in self.config.columns.values()]
```

Comment:
I am not sure I understand why there are potentially more than a single column considered here. I would have expected a single column for the flux measurement and that this would be selected with the `flux` key in the config.

Reply:
There's no guarantee that the user will name the key in their input dict using the key `flux`. They might easily choose another keyword that best describes the flux data they want to generate a metric for. As the keyword may vary, we had two options: either enforce a keyword constraint by making the case in the doc-string that only the keyword `flux` will be used, or use this simple list comprehension to extract the columns of interest. As the keyword choice is largely irrelevant here, I opted for the simpler latter approach.

Reply:
Thanks, that makes sense.